src/Graphs/DirectedAcyclicGraphTransitiveReducer.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Bearded.Utilities.Graphs
+{
+    public static class DirectedAcyclicGraphTransitiveReducer<T> where T : IEquatable<T>
+    {
+        public static IDirectedAcyclicGraph<T> ReduceGraph(IDirectedAcyclicGraph<T> graph)
+        {
+            var elements = graph.Elements as IList<T> ?? graph.Elements.ToList();
+            var graphBuilder = DirectedGraphBuilder<T>.NewBuilder();
+
+            foreach (var element in elements)
+            {
+                graphBuilder.AddElement(element);
+            }
+
+            foreach (var element in elements)
+            {
+                foreach (var neighbor in getOnlyDirectChildren(graph, element))
```

Comment:
I'd expect the iterator to be called `child`, given the name of the method.

Reply:
This is caused by a previous naming. Will update